README.md

```diff
@@ -1,548 +1,309 @@
+### Substrate
+#### Deposit
+You can call the deposit function from the substrate pallet.
+You don't need separately calculate the fees or send approval transaction before the deposit.
+```typescript
+// more code in example above
+const unsub = await deposit(api, asset, amount, domainId, address)
+  .signAndSend(currentAccount.address, { signer: injector.signer }, result => {
+     handleTxExtrinsicResult(api, result, unsub, {
+      onDepositEvent: (depositData) => {
+        // Log the deposit nonce (unique identifier for the transfer)
+        console.log("depositNonce", depositData.depositNonce)
       },
-    ],
-  },
-];
+     });
+   });
 ```
-
-Then, inside your App, create some state variables and functions to get account data from your wallet (in this example the wallet is `Metamask`)
-
-```ts
-function App(){
-  const [bridge, setBridge] = useState<any | undefined>(undefined)
-
-  const [logicConnected, setLogicConnected] = useState<boolean>(false);
-
-  const [sygmaInstance, setSygmaInstance] = useState<Sygma | undefined>(undefined);
-
-  const getAccountData = async (sygma: Sygma) => {
-    try {
-      const balance =
-        (await sygma.getSignerBalance("chain1")) ?? BigNumber.from("0");
-      const address = await sygma.getSignerAddress("chain1");
-      const gasPrice = await sygma.getSignerGasPrice("chain1");
-      const { balanceOfTokens, tokenName } = await sygma.getTokenInfo(
-        "chain1"
-      );
-      console.log("signer balance", utils.formatEther(balance!));
-      console.log("signer address", address);
-      console.log("gas price", utils.formatEther(gasPrice!));
-      console.log("balance of tokens", utils.formatUnits(balanceOfTokens, 18));
-      setValue("address", address!)
-      setAccountData({
-        balance: balance!,
-        address: address!,
-        gasPrice: gasPrice!,
-        balanceOfTokens: balanceOfTokens!,
-        tokenName: tokenName!,
-      });
-      setIsReady(true);
-    } catch (e) {
-      console.log(e);
-      console.log("Perhaps you forget to deploy the bridge?");
-    }
-  };
-
-  useEffect(() => {
-    if (data !== undefined && sygmaInstance !== undefined) {
-      getAccountData(sygmaInstance);
-    }
-  }, [data, logicConnected]);
-
-  useEffect(() => {
-    if (metaIsConnected && sygmaInstance !== undefined) {
-      handleConnect();
-      getAccountData(sygmaInstance! as Sygma);
-      // Update form values using useForm
-      setValue("from", sygmaInstance.bridgeSetup?.chain1.domainId!)
-      setValue("to", sygmaInstance.bridgeSetup?.chain2.domainId!)
-    }
-  }, [metaIsConnected]);
-
-}
+#### Substrate network as destination
+Using deposit nonce you can listen for proposal execution event on substrate side in case of transfering from EVM network:
+
+```typescript
+import { listenForEvent } from '@buildwithsygma/sygma-sdk-core/Substrate'
+// Assuming you have a valid ApiPromise instance (api)
```

Comment:
to improve understandability of this example, we should provide again how we instance `ApiPromise` instance from dot